### PR TITLE
Migration to libusb1 and changes to make it compatible with Python on Windows

### PR DESCRIPTION
**Pull request by juanmitaboada**
… Windows

### Contributor checklist

- [ ] I have read the CONTRIBUTING.rst
- [ ] I have tested my contribution on these devices:
 * e.g. Epson TM-T88II
- [ ] My contribution is ready to be merged as is

----------

### CODE REVIEW

**Comment by patkan** [comment]
Hi, thank you for your contribution :-) I'm going to have a look at it later.
In order to make the AUTHORS work like you added it, you could use a mailmap line like this:
```
Juanmi Taboada // br0th3r <juanmi@juanmitaboada.com> Juanmi Taboada <juanmi@centrologic.com>
```

**Comment by br0th3r** [comment]
Thank you, AUTHORS corrected. :-)

File: src/escpos/printer.py [hunk 1]
```diff
@@ -52,47 +54,38 @@ def __init__(self, idVendor, idProduct, timeout=0, in_ep=0x82, out_ep=0x01, *arg
...
-        except NotImplementedError:
-            pass
+        self.device = dev.open()
+        if platform.system() == 'Linux' and self.device.kernelDriverActive(self.INTERFACE):
+            self.device.detachKernelDriver(self.INTERFACE)
 
-        if check_driver is None or check_driver:
-            try:
-                self.device.detach_kernel_driver(0)
-            except usb.core.USBError as e:
```
Comment (patkan) [inline src/escpos/printer.py]: Would it be possible to also check for this exception in your code?

File: setup.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def run_tests(self):
         'Topic :: Office/Business :: Financial :: Point-Of-Sale',
     ],
     install_requires=[
-        'pyusb>=1.0.0',
+        'libusb1',
```
Comment (patkan) [inline setup.py]: Are all versions of this lib OK?

Comment (br0th3r) [reply setup.py]: I haven't checked other versions of libusb1.

File: src/escpos/printer.py (hunk 1 above)
Comment (br0th3r) [reply src/escpos/printer.py]: I am not sure if this is necesary with libusb1.

**Comment by br0th3r** [comment]
Pyusb didn't work on Windows for me after 10h trying on several ways. Reading on Internet I got to know that there several differences between the API v0.1 and v1.x of libusb1 and Pyusb is not compatible with the latest which happens to work properly on Windows.

I have checked this library on Linux and Windows 7 for an EPSON TM-T20II (MODEL M267D) and it worked like a charm. :-)

**Comment by patkan** [comment]
So if nobody opposes I would merge this in the coming days.

**Comment by belono** [comment]
Is possible to extend support for libusb1 without dropping pyusb?

**Comment by patkan** [comment]
I will research how to set optional dependencies for a Python package. This way we could use both libraries.

What bothers me a bit now is that I think I remember that pyusb allegedly supports libusbv1.
I will also research this.

**Comment by patkan** [comment]
It seems possible to specify platform-dependant requirements. https://stackoverflow.com/a/32955538/4244236 https://stackoverflow.com/a/42946495/4244236
This change should keep the old library an allow to switch to the new dependency. Ideally the proper package is preselected for the platform and can be switched. (In the module this can be realized by trying to import the first one, then falling back to the other if the import does not succeed.)

**Comment by om26er** [comment]
@patkan @belono I made an attempt to make things work on Windows here https://github.com/python-escpos/python-escpos/pull/330

**Pull request closed without merging**